imap_processing/codice/codice_l1a.py

```
@@ -366,17 +290,14 @@ def process_codice_l1a(packets) -> str:
         elif apid == CODICEAPID.COD_LO_SW_SPECIES_COUNTS:
             print("Processing COD_LO_SW_SPECIES_COUNTS packet")
 
-            # These are currently commented out because the test data does not
-            # have SHCOARSE
-            # TODO: Turn these "on" once proper testing data is acquired
-            # packets = grouped_data[apid]
-            # sorted_packets = sort_by_time(packets, "SHCOARSE")
+            packets = sort_by_time(grouped_data[apid], "SHCOARSE")
 
             # Get the four "main" parameters for processing
-            table_id, plan_id, plan_step, view_id = get_params(apid)
+            table_id, plan_id, plan_step, view_id = get_params(packets[0])
 
             # Run the pipeline to create a dataset for the product
             pipeline = CoDICEL1aPipeline(table_id, plan_id, plan_step, view_id)
+            pipeline.packets = packets
```

If user forget to initialized `pipeline.packets` here, then looks like `pipeline.unpack_science_data()` and `pipline.create_science_dataset()` will fail since it's not in the init function. Can you add a check and may be raise value missing error?

---

Good catch!

---

On second thought, I think it makes sense to pass in the `packets` as a parameter to those two methods, so I will do that.